### PR TITLE
added nextcloud integration research

### PR DESCRIPTION
**Pull request by absurd-turtle**
Summary of my research regarding the Nextcloud API capabilities

### CODE REVIEW

File: doc/research/nextcloud_integration.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+The circle app has a REST API but it is not documented. The routes of the API can be found in
+https://github.com/nextcloud/circles/blob/22238597fb9045e748119247fceaac7321f0a31e/appinfo/routes.php
+
+API test with curl (USERNAME, PSWD and BASE_URL have to set):
+```
+curl -u USERNAME:PSWD -X GET 'https://BASE_URL/nextcloud/ocs/v2.php/apps/circles/circles' -H "OCS-APIRequest: true"
+```
+This returns a list of circles visible to the authenticated user.
+
+Tested API calls:
```
Comment (markus2330) [inline doc/research/nextcloud_integration.md:48]: I assume all of them work?

```suggestion
Working API calls (tested with NC version 25.0.4-1~deb11):

```

File: doc/research/nextcloud_integration.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+- https://github.com/nextcloud/contacts
+- https://github.com/nextcloud/contacts/blob/main/lib/Dav/PatchPlugin.php
+- [CardDAV clients](https://devguide.calconnect.org/CardDAV/Client-Implementations/)
+- [CardDAV libraries](https://devguide.calconnect.org/CardDAV/libraries/)
+- https://github.com/nextcloud/circles
+- [Circle routes](https://github.com/nextcloud/circles/blob/22238597fb9045e748119247fceaac7321f0a31e/appinfo/routes.php)
+
+
+nextcloud/contacts is based on [sabredav](https://github.com/sabre-io/dav)(most popular WebDAV framework for PHP)
+
```
Comment (markus2330) [inline doc/research/nextcloud_integration.md:88]:
````suggestion

## Chat

sending a message:

```rust
// sends once, Err if it does not work on network or nextcloud level
fn send_message (
	&self,
	message: String,
	chat: String,
) -> Result<reqwest::Response, reqwest::Error> {
	let mut headers = header::HeaderMap::new();
	headers.insert("Content-Type", "application/json".parse().unwrap());
	headers.insert("Accept", "application/json".parse().unwrap());
	headers.insert("OCS-APIRequest", "true".parse().unwrap());

	let result = reqwest::Client::new()
		.post(&format!(
			"{}/ocs/v2.php/apps/spreed/api/v1/chat/{}",
			&self.url, chat
		))
		.basic_auth(&self.user, self.pass.as_ref())
		.headers(headers)
		.body(format!(
			"{{\"token\": \"{}\", \"message\": \"{}\"}}",
			chat, message
		))
		.send();
	match result {
		Ok(response) => match response.error_for_status() {
			Ok(response) => Ok(response),
			Err(error) => Err(error),
		},
		Err(error) => Err(error),
	}
}
```

or via CURL:
```
curl -sS -d "{\"token\": \"$CHAT\", \"message\": \"$*\"}" -H "Content-Type: application/json" -H "Accept: application/json" -H "OCS-APIRequest: true" -u "user:password" https://nextcloud.markus-raab.org/nextcloud/ocs/v2.php/apps/spreed/api/v1/chat/$CHAT
```
````

**Review by markus2330 — APPROVED** [review]
Great investigations!

File: doc/research/nextcloud_integration.md (hunk 1 above)
Comment (absurd-turtle) [reply doc/research/nextcloud_integration.md:48]: Yes

**Pull request merged**